### PR TITLE
Formatting retains comments (module header and paren expressions)

### PR DESCRIPTION
**Pull request by avh4**
This is an incremental part of the retain comments task of https://github.com/gren-lang/compiler/issues/42.  More locations of comments will be retained in future PRs (those remaining locations are being tracked by compiler warnings).

This implements comment handling in the following places:
- before the `module` line
- `module 💬 Name 💬 exposing 💬 (..) 💬`
- between `module` line and module doc comments
- between module doc comments and `import`s
- `(💬 expression)`

This also sets up a pattern for testing formatting behavior.

### CODE REVIEW

**Comment by avh4** [comment]
@robinheghan This sets up the patterns for how we'll retain comments and how we'll test formatting.  I think with this set I can start doing small incremental PRs to add more comment handling.  So this one I think is ready for review, and let's especially discuss anything about the code patterns before I continue doing more.

File: compiler/src/AST/Source.hs [hunk 1]
```diff
@@ -176,14 +170,14 @@ data Port = Port (A.Located Name) Type
 
 data Effects
   = NoEffects
-  | Ports [(SourceOrder, Port)]
-  | Manager A.Region Manager
+  | Ports [(SourceOrder, Port)] SC.PortsComments
+  | Manager A.Region Manager SC.ManagerComments
```
Comment (avh4) [inline compiler/src/AST/Source.hs:174]: 💭 another option would be to have these import as `Comments.Manager` (having the type name `Manager` overloaded but expect that it's always used via qualified import)

File: compiler/src/Reporting/Error/Syntax.hs [hunk 2]
```diff
@@ -70,6 +70,7 @@ data Error
   | NoPortModulesInPackage A.Region
   | NoEffectsOutsideKernel A.Region
   | ParseError Module
+  deriving (Show)
```
Comment (avh4) [inline compiler/src/Reporting/Error/Syntax.hs:73]: Show instances in this file aren't actually needed for anything yet, but they were helpful when debugging.

File: gren.cabal [hunk 3]
```diff
@@ -249,8 +250,9 @@ Test-Suite gren-tests
         Helpers.Parse
 
         -- tests
-        Parse.SpaceSpec
+        Integration.FormatSpec
         Parse.RecordUpdateSpec
+        Parse.SpaceSpec
```
Comment (avh4) [inline gren.cabal:255]: sorted this section alphabetically

File: compiler/src/AST/Source.hs (hunk 1 above)
Comment (robinheghan) [reply compiler/src/AST/Source.hs:174]: Importing as `Comments.Manager` seems like a better option to me, but I'm fine with how it looks now as well.

File: compiler/src/Reporting/Error/Syntax.hs (hunk 2 above)
Comment (robinheghan) [reply compiler/src/Reporting/Error/Syntax.hs:73]: Yeah. My bar for adding `Show` is pretty low. I spend enough time debugging that it feels I'm often adding these all the time.

Leave it in 👍

**Review by robinheghan — APPROVED** [review]
I think this looks good =)

File: compiler/src/AST/Source.hs (hunk 1 above)
Comment (avh4) [reply compiler/src/AST/Source.hs:174]: 👍 I'll save considering this for a future PR, then

**Comment by avh4** [comment]
> I think this looks good =)

@robinheghan  There appears to be no "Merge" button for me even after this is approved, so go ahead an merge whenever.

**Pull request merged**